### PR TITLE
Feat: Searching for an array of tokens in asks

### PR DESCRIPTION
**Pull request by Tevzi2**
Implements the ability to search for an array of tokens instead of a single token in orders/asks, proposition is implemented as v5 of orders asks.

### CODE REVIEW

File: packages/indexer/src/api/endpoints/orders/get-orders-asks/v5.ts [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+  },
+  validate: {
+    query: Joi.object({
+      ids: Joi.alternatives(Joi.array().items(Joi.string()), Joi.string()).description(
+        "Order id(s) to search for."
+      ),
+      tokens: Joi.alternatives()
+        .try(
+          Joi.array().items(Joi.string().lowercase().pattern(regex.token)),
+          Joi.string().lowercase().pattern(regex.address)
```
Comment (ipeleg) [inline packages/indexer/src/api/endpoints/orders/get-orders-asks/v5.ts:52]: should be regex.token, same as the pattern for the array items

File: packages/indexer/src/api/endpoints/orders/get-orders-asks/v5.ts [hunk 2]
```diff
@@ -0,0 +1,571 @@
...
+          conditions.push(`orders.id IN ($/ids:csv/)`);
+        } else {
+          conditions.push(`orders.id = $/ids/`);
+        }
+      } else {
+        orderStatusFilter = `orders.fillability_status = 'fillable' AND orders.approval_status = 'approved'`;
+      }
+
+      if (query.tokens) {
+        if(!_.isArray(query.tokens)) {
```
Comment (ipeleg) [inline packages/indexer/src/api/endpoints/orders/get-orders-asks/v5.ts:283]: Shouldn't this if check be `if(_.isArray(query.tokens))` ? if not the map inside the if will fail as it's not an array according to the current check

File: packages/indexer/src/api/endpoints/orders/get-orders-asks/v5.ts [hunk 3]
```diff
@@ -0,0 +1,571 @@
...
+          (query as any).tokensPrefix = [query.tokens].map(token => `token:${token}`);
+        }
+
+        (query as any).tokensFilter = query.tokensPrefix.map(toBuffer);
```
Comment (ipeleg) [inline packages/indexer/src/api/endpoints/orders/get-orders-asks/v5.ts:287]: `query.tokensPrefix` can be undefined hence the .map will fail

Comment (ipeleg) [inline packages/indexer/src/api/endpoints/orders/get-orders-asks/v5.ts:287]: no need to to toBuffer as token_set_id in the orders table is a string and not a byte type column

**Review by ipeleg — CHANGES_REQUESTED** [review]

File: packages/indexer/src/api/endpoints/orders/get-orders-asks/v5.ts (hunk 2 above)
Comment (Tevzi2) [reply packages/indexer/src/api/endpoints/orders/get-orders-asks/v5.ts:283]: My bad wrote what happens inside of the if statement incorrectly, if tokens is not an array => convert it to an array, then map

**Pull request closed without merging**